subcommand/tls-init/commant_test.go

```diff
@@ -0,0 +1,591 @@
+	certBlock, _ := pem.Decode(newServerCert)
+	certificate, err := x509.ParseCertificate(certBlock.Bytes)
+	require.NoError(t, err)
+	require.False(t, certificate.IsCA)
+	require.Equal(t, []string{"test.dns.name", "server.dc1.consul", "localhost"}, certificate.DNSNames)
+
+	require.NotEqual(t, serverKey, newServerKey)
+	keyBlock, _ := pem.Decode(newServerKey)
+	privateKey, err := x509.ParseECPrivateKey(keyBlock.Bytes)
+	require.NoError(t, err)
+	require.Equal(t, &privateKey.PublicKey, certificate.PublicKey)
+}
+
+func TestRun_CreatesServerCertificatesWithExistingCertsAsSecrets(t *testing.T) {
+	ui := cli.NewMockUi()
+	cmd := Command{UI: ui}
+	k8s := fake.NewSimpleClientset()
+
+	_, err := k8s.CoreV1().Secrets("default").Create(context.Background(), &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "consul-ca-cert",
+			Namespace: "default",
+		},
+		Data: map[string][]byte{
+			corev1.TLSCertKey: []byte(caCert),
+		},
+		Type: corev1.SecretTypeOpaque,
+	}, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	_, err = k8s.CoreV1().Secrets("default").Create(context.Background(), &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "consul-ca-key",
+			Namespace: "default",
+		},
+		Data: map[string][]byte{
+			corev1.TLSPrivateKeyKey: []byte(caKey),
+		},
+		Type: corev1.SecretTypeOpaque,
+	}, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	cmd.clientset = k8s
+
+	flags := []string{"-name-prefix", "consul"}
+
+	exitCode := cmd.Run(flags)
+	require.Equal(t, 0, exitCode)
+
+	serverCertSecret, err := k8s.CoreV1().Secrets("default").Get(context.Background(), "consul-server-cert", metav1.GetOptions{})
+	require.NoError(t, err)
+	serverCert := serverCertSecret.Data[corev1.TLSCertKey]
+	serverKey := serverCertSecret.Data[corev1.TLSPrivateKeyKey]
+
+	certBlock, _ := pem.Decode([]byte(serverCert))
+	certificate, err := x509.ParseCertificate(certBlock.Bytes)
+	require.NoError(t, err)
+	require.False(t, certificate.IsCA)
+	require.Equal(t, []string{"server.dc1.consul", "localhost"}, certificate.DNSNames)
+	require.Equal(t, []net.IP{net.ParseIP("127.0.0.1").To4()}, certificate.IPAddresses)
+
+	keyBlock, _ := pem.Decode([]byte(serverKey))
+	privateKey, err := x509.ParseECPrivateKey(keyBlock.Bytes)
+	require.NoError(t, err)
+	require.Equal(t, &privateKey.PublicKey, certificate.PublicKey)
+}
+
+func TestRun_CreatesServerCertificatesWithoutExistingCerts(t *testing.T) {
+	ui := cli.NewMockUi()
+	cmd := Command{UI: ui}
+	k8s := fake.NewSimpleClientset()
+	cmd.clientset = k8s
+
+	flags := []string{"-name-prefix", "consul"}
+	exitCode := cmd.Run(flags)
+	require.Equal(t, 0, exitCode)
+
+	serverCertSecret, err := k8s.CoreV1().Secrets("default").Get(context.Background(), "consul-server-cert", metav1.GetOptions{})
+	require.NoError(t, err)
+	newServerCert := serverCertSecret.Data[corev1.TLSCertKey]
+	newServerKey := serverCertSecret.Data[corev1.TLSPrivateKeyKey]
+
+	require.NotEqual(t, serverCert, newServerCert)
```

Comment:
does `serverCert` here exist from a previous test? Would we want to explicitly create it as empty here instead? (Ignore this if its not test pollution :) )

Reply:
This is definitely a holdover from another test. Excellent catch. Gonna check these changes in now.